android/app/src/main/java/com/sample/MainActivity.kt

```diff
@@ -36,7 +38,7 @@ class MainActivity : ReactActivity() {
 
         val reactNativeHost = reactNativeHost
         if (reactNativeHost is TestAppReactNativeHost) {
-            reactNativeHost.currentActivity = this
+            reactNativeHost.startInBackground()
```

Comment:
It will crash after configuration change (rotate the phone, for example) because of the assertion. We should check if the instance is already started before calling `startInBackground()` here.